File: src/main/kotlin/ui/screens/UpdateSlipSingle.js
import React, { useEffect, useState } from 'react';

const UpdateSlipSingle = (params) => {
  const [number, setNumber] = useState(params.number);
  const [amount, setAmount] = useState(params.amount);

  return (
    <form>
      <h2 className="text-2xl font-semibold mb-4">Update Slip</h2>
      <div className="flex items-center gap-4 mb-4">
        <input
          type="text"
          value={number}
          onChange={(e) => setNumber(e.target.value)}
          placeholder="Number"
          className="flex-1 p-2 border border-gray-300 rounded"
        />
        <input
          type="text"
          value={amount}
          onChange={(e) => setAmount(e.target.value)}
          placeholder="Amount"
          className={`flex-1 p-2 border border-gray-300 rounded`}
        />
        <button
          type="button"
          onClick={() => {
            params.handleUpdate(number, amount)
            setNumber('');
            setAmount('');
          }}
          className="p-2 bg-blue-500 text-white rounded hover:bg-blue-600"
        >
          Update
        </button>
      </div>
    </form>
  );
};

export default UpdateSlipSingle;

File: src/main/kotlin/ui/screens/UpdateSlipSingle.kt
package ui.screens

import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.foundation.text.KeyboardOptions
import core.config.CompactOutlinedTextField

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun UpdateSlipSingle(
    initialNumber: String = "",
    initialAmount: String = "",
    onUpdate: (String, String) -> Unit
) {
    var number by remember { mutableStateOf(initialNumber) }
    var amount by remember { mutableStateOf(initialAmount) }

    // Update state when initial values change
    LaunchedEffect(initialNumber, initialAmount) {
        number = initialNumber
        amount = initialAmount
    }

    val handleUpdate = {
        onUpdate(number, amount)
        number = ""
        amount = ""
    }

    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(16.dp)
    ) {
        // Title
        Text(
            text = "Update Slip",
            fontSize = 24.sp,
            fontWeight = FontWeight.SemiBold,
            modifier = Modifier.padding(bottom = 16.dp)
        )

        // Input fields and button in a row
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(bottom = 16.dp),
            horizontalArrangement = Arrangement.spacedBy(16.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            // Number input field
            CompactOutlinedTextField(
                value = number,
                onValueChange = { number = it },
                label = { Text("Number") },
                placeholder = { Text("Number") },
                modifier = Modifier.weight(1f),
                keyboardOptions = KeyboardOptions(
                    keyboardType = KeyboardType.Text
                )
            )

            // Amount input field
            CompactOutlinedTextField(
                value = amount,
                onValueChange = { amount = it },
                label = { Text("Amount") },
                placeholder = { Text("Amount") },
                modifier = Modifier.weight(1f),
                keyboardOptions = KeyboardOptions(
                    keyboardType = KeyboardType.Number
                )
            )

            // Update button
            Button(
                onClick = handleUpdate,
                modifier = Modifier.height(56.dp),
                colors = ButtonDefaults.buttonColors(
                    containerColor = MaterialTheme.colorScheme.primary
                )
            ) {
                Text(
                    text = "Update",
                    fontWeight = FontWeight.Medium
                )
            }
        }
    }
}